specs/error_state/ErrorOutOfGasEXP.md

```diff
@@ -0,0 +1,31 @@
+# ErrorOutOfGasEXP state for EXP OOG error
+
+## Procedure
+
+Handle the corresponding out of gas error for `EXP`.
+
+### EVM behavior
+
+The EXP gas cost is calculated as:
+
+```
+# `exponent.bits()` returns the least number of bits needed to represent exponent.
+exponent_byte_size = (exponent.bits() + 7) // 8
+
+gas_cost = exponent_byte_size * 50 + 10
+```
+
+### Constraints
+
+1. Constrain `gas_left < gas_cost`.
+2. Current call must fail.
+3. If it's a root call, it transits to `EndTx`.
+4. If it isn't a root call, it restores caller's context by reading to `rw_table`, then does step state transition to it.
+5. Constrain `rw_counter_end_of_reversion = rw_counter_end_of_step + reversible_counter`.
+
+### Lookups
+
+3 bus-mapping lookups:
+
+1. 1 stack read for the second stack pop value `exponent` which is only related with gas cost calculation.
+2. 2 call context lookups for `is_success` and `rw_counter_end_of_reversion`.
```

Comment:
add RestoreContext lookups .

Reply:
Add the `restore context lookups` for non-root call to this Markdown doc in [3a1b1fe](https://github.com/scroll-tech/zkevm-specs/pull/58/commits/3a1b1fef7080732b2a938573c3e3aec46dfd4025).